kubetest2-gke/deployer/firewall.go

```diff
@@ -90,25 +95,25 @@ func clusterFirewallName(project, cluster string, instanceGroups map[string]map[
 // Ensure firewall rules for multi-project profile.
 // Reference: https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-shared-vpc#creating_additional_firewall_rules
 // Please note we are not including the firewall rule for SSH connection as it's not needed for testing.
-func ensureFirewallRulesForMultiProjects(projects []string, network string, subnetworkRanges []string) error {
-	hostProject := projects[0]
+func (d *Deployer) ensureFirewallRulesForMultiProjects() error {
+	hostProject := d.Projects[0]
 	hostProjectNumber, err := getProjectNumber(hostProject)
 	if err != nil {
 		return fmt.Errorf("error looking up project number for id %q: %w", hostProject, err)
 	}
-	for i := 1; i < len(projects); i++ {
-		curtProject := projects[i]
+	for i := 1; i < len(d.Projects); i++ {
+		curtProject := d.Projects[i]
 		curtProjectNumber, err := getProjectNumber(curtProject)
 		if err != nil {
 			return fmt.Errorf("error looking up project number for id %q: %w", curtProject, err)
 		}
 		firewall := fmt.Sprintf("rule-%s-%s", hostProjectNumber, curtProjectNumber)
 		// sourceRanges need to be separated with ",", while the provided subnetworkRanges are separated with space.
-		sourceRanges := strings.ReplaceAll(subnetworkRanges[i-1], " ", ",")
+		sourceRanges := strings.ReplaceAll(d.SubnetworkRanges[i-1], " ", ",")
 		if err := runWithOutput(exec.Command("gcloud", "compute", "firewall-rules", "create", firewall,
 			"--project="+hostProject,
-			"--network="+network,
-			"--allow=tcp,udp,icmp",
+			"--network="+d.Network,
+			"--allow="+d.FirewallRuleAllow,
```

Comment:
I assume the ports syntax also works here

https://cloud.google.com/sdk/gcloud/reference/compute/firewall-rules/create#--allow